### PR TITLE
feat: execute report on frontend

### PR DESCRIPTION
**Pull request by Ensar025**

### CODE REVIEW

File: packages/financial-health-calculator/src/data-fetchers/open-collective.ts [hunk 1]
```diff
@@ -56,6 +56,7 @@ export const fetchOpenCollectiveData = (packageName: string): Promise<OpenCollec
 
   return fetch('https://api.opencollective.com/graphql/v2', options)
     .then((response) => response.json())
+    .catch((err) => console.log(err))
```
Comment (vanhoofmaarten) [inline packages/financial-health-calculator/src/data-fetchers/open-collective.ts:59]: Choose either a `console.error` or

```js
.catch(error => {
    throw new Error(error);
})
```

File: packages/financial-health-calculator/src/helpers/number-formatter.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+export const formatCurrency = (number: number, currency: string) => {
+  return new Intl.NumberFormat('en-GB', { style: 'currency', currency }).format(number);
```
Comment (vanhoofmaarten) [inline packages/financial-health-calculator/src/helpers/number-formatter.ts:2]: Try to get the user's preferred locale via the Accept-Language header

**Review by vanhoofmaarten — CHANGES_REQUESTED** [review]

File: packages/financial-health-calculator/src/data-fetchers/open-collective.ts (hunk 1 above)
Comment (Ensar025) [reply packages/financial-health-calculator/src/data-fetchers/open-collective.ts]: Resolved in e0471b3c7d649d657e6d0f4e559786ab6954f785

File: packages/financial-health-calculator/src/helpers/number-formatter.ts (hunk 2 above)
Comment (Ensar025) [reply packages/financial-health-calculator/src/helpers/number-formatter.ts]: Resolved in ec2df43d7ab634ed0eb49fd485433e0b3ef7d46b

**Review by vanhoofmaarten — CHANGES_REQUESTED** [review]

File: packages/backend/financial-health-report/index.mts [hunk 3]
```diff
@@ -24,7 +25,15 @@ const httpTrigger: AzureFunction = async function (context: Context, req: HttpRe
   }
 
   const packageName = req.body.packageName;
-  const financialHealthReport = await calculateFinancialHealthScore(packageName);
+
+  let localeString = 'en-GB';
+
+  if (req.headers['accept-language']) {
+    const localeObject: parser.Language = parser.parse(req.headers['accept-language'])[0];
+    localeString = localeObject.code + localeObject.region || '';
```
[the remark this answers is not in the document]
Comment (Ensar025) [reply packages/backend/financial-health-report/index.mts:33]: > Aren't we missing a `-`?

Yes we do



> I suggest to use the following pattern: `[localeObject.code, localeObject.region].filter(str => !!str).join('-')`
> 
> Put both values in an array, filter out undefined values and join multiple values as a string divivded by a `-`
> 
> ```js
> ['en', 'GB'].filter(str => !!str).join('-')
> // 'GB'
> ['en', undefined].filter(str => !!str).join('-')
> // 'en'
> [undefined, undefined].filter(str => !!str).join('-')
> // ''
> ```

Yes this implementation is better

[the remark this answers is not in the document]
Comment (Ensar025) [reply packages/backend/financial-health-report/index.mts]: Resolved in 435a06c2f72f9beeb5f40a09f97536f1f4d85c9f

**Review by vanhoofmaarten — APPROVED** [review]

**Pull request merged**